src/widgets/scrollbar.rs

```diff
@@ -1128,4 +1131,341 @@ mod tests {
             assert_buffer_eq!(buffer, Buffer::with_lines(expected.clone()));
         }
     }
+
+    #[rstest]
+    #[case("██", 0, 2, "position_0")]
+    #[case("██", 1, 2, "position_1")]
+    fn render_scrollbar_simplest(
+        #[case] expected: &str,
+        #[case] position: usize,
+        #[case] content_length: usize,
+        #[case] assertion_message: &str,
+    ) {
+        let size = expected.width() as u16;
+        let mut buffer = Buffer::empty(Rect::new(0, 0, size, 1));
+        let mut state = ScrollbarState::default()
+            .position(position)
+            .content_length(content_length);
+        Scrollbar::default()
+            .orientation(ScrollbarOrientation::HorizontalBottom)
+            .begin_symbol(None)
+            .end_symbol(None)
+            .render(buffer.area, &mut buffer, &mut state);
+        assert_eq!(
+            buffer,
+            Buffer::with_lines(vec![expected]),
+            "{}",
+            assertion_message
+        );
+    }
+
+    #[rstest]
+    #[case("##########", 0, 10, "position_0")]
+    #[case("##########", 1, 10, "position_1")]
+    #[case("##########", 2, 10, "position_2")]
+    #[case("##########", 3, 10, "position_3")]
+    #[case("##########", 4, 10, "position_4")]
+    #[case("##########", 5, 10, "position_5")]
+    #[case("##########", 6, 10, "position_6")]
+    #[case("##########", 7, 10, "position_7")]
+    #[case("##########", 8, 10, "position_8")]
+    #[case("##########", 9, 10, "position_9")]
```

Comment:
Maybe a comment here (and for the above test) to indicate why the scrollbar is completely filled. It confused me at first and it took me some time to figure it out.

Reply:
For context this is what the old implementation does, so it is merely a "specification test". 

This entire PR is just to show how the new tests behave in the old implementation. I'm planning to revert the commit after this PR is merged to use the new tests.

Here's what the new tests do here:

<img width="320" alt="image" src="https://github.com/ratatui-org/ratatui/assets/1813121/077c4b4b-66e6-498b-a561-98caf36a2fc9">

Reply:
So is this perhaps due to the fact that content length and viewport are about the same size? If so, need to add more tests where that isn't the case.